tests/unit_tests/fn/test_fn_scan.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2021, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#include <gridtools/fn/scan.hpp>
+
+#include <gtest/gtest.h>
+
+#include <gridtools/sid/composite.hpp>
+#include <gridtools/sid/synthetic.hpp>
+
+namespace gridtools::fn {
+    namespace {
+        using namespace literals;
+        using sid::property;
+
+        struct sum_fold : fwd {
+            static GT_FUNCTION constexpr auto body() {
+                return [](auto acc, auto const &iter) { return acc + *iter; };
+            }
+        };
+
+        struct sum_scan : fwd {
+            static GT_FUNCTION constexpr auto body() {
+                return scan_pass(
+                    [](auto acc, auto const &iter) {
+                        return tuple_util::host_device::make<tuple>(get<0>(acc) + *iter, get<1>(acc) * *iter);
```

Comment:
as soon it is already c++17,  `tuple_util::host_device::make<tuple>`  can be replaced by just `tuple` here and below

Reply:
Oh man, I thought that was in 20, too many versions there :P